DatadogRUM/Sources/RUMMonitor/Scopes/RUMApplicationScope.swift

```diff
@@ -104,6 +105,10 @@ internal class RUMApplicationScope: RUMScope, RUMContextProvider {
                 return scope
             }
 
+            // proccss(command:context:writer) returned false, so the scope will be deallocated at the end of
+            // this execution context. End the "RUM Session Ended" metric:
+            defer { dependencies.sessionEndedMetric.endMetric(sessionID: scope.sessionUUID) }
+
```

Comment:
If I understand correctly, the `endMetric` can be triggered only when there is a command to process and it returns false, otherwise it can not be triggered it self even if the duration and timeout has passed the limit?

Reply:
That is correct 👍. Every state change in RUM is caused by command. Even in case of "15min timeout" or "4h max duration", the `RUMApplicationScope` only recognizes these upon receiving next command.